.github/workflows/docsbuild-release.yaml

```diff
@@ -0,0 +1,84 @@
+name: ansible-release-porting-guide
+on:
+  workflow_dispatch:
+    inputs:
+      ansible_version:
+        description: 'Release Version. Example : 11.1.0'
+        required: true
+
+env:
+  CI_COMMIT_MESSAGE: 'Add Ansible community "${{ inputs.ansible_version }}" porting guide'
+
+
+jobs:
+  build:
+    name: 'Build Ansible community distribution (${{ inputs.ansible_version }})'
+    runs-on: ubuntu-latest
+
+    steps:
+
+# Get the major version from the user input
+
+      - name: Retrieve the major version component
+        id: compute-version
+        run: |
+          echo -n 'major-version=' >> "${GITHUB_OUTPUT}"
+          echo '${{ inputs.ansible_version }}' | sed 's#\([0-9]\+\)\..*#\1#' >> "${GITHUB_OUTPUT}"
+
+# checkout the /ansible/ansible-documentation & ansible-build-data
+      - name: Check out ansible-build-data under antsibull build directory
+        uses: actions/checkout@v3
+        with:
+          path: /ansible/ansible-documentation
+          ref: test
+
+      - name: Check out ansible-build-data under antsibull build directory
+        uses: actions/checkout@v3
+        with:
+          path: antsibull/build/ansible-build-data
+          ref: test
+
+      - name: Set up Python 3.11
+        uses: actions/checkout@v4
+        with:
+          repository: ansible-community/ansible-build-data
+          ref: ${{ inputs.ansible-version }}
+          sparse-checkout: ${{ steps.compute-version.outputs.major-version }}/porting_guide_${{ steps.compute-version.outputs.major-version }}.rst
+
+      - name: Checking out to a new branch & setting the user details
+        run: |
+          cd antsibull/build/ansible-build-data
+          git checkout -b "publish-${{ inputs.ansible_version }}"
+          git config --global user.name "${{github.actor}}"
+          git config --global user.email "${{github.actor}}@users.noreply.github.com"
+
+      - name: Commit ansible-build-data and push the changes to github
+        run: |
+          pwd
+          wget https://files.pythonhosted.org/packages/source/a/ansible/"ansible-${{ inputs.ansible_version }}".tar.gz
+          tar -xvf ansible-${{ inputs.ansible_version }}".tar.gz
+          cp "ansible-${{ inputs.ansible_version }}"/"porting_guide_${{ inputs.ansible_version }}.rst" ./ansible-documentation/docs/docsite/rst/porting_guides/
+
+      - name: Commit ansible-documenatation  and push the changes to github
+        run: |
+         cd ./ansible-documentation
+          git add ./docs/docsite/rst/porting_guides/"porting_guide_${{ inputs.ansible_version }}.rst"
+          git commit -m "${{ env.CI_COMMIT_MESSAGE }}"
+          git push origin "publish-${{ inputs.ansible_version }}"
```

Comment:
How about making the push a separate step and using this action? https://github.com/ad-m/github-push-action/tree/v0.8.0/

It would be something like:

```
- name: Push porting guide to the docs repository
   uses: ad-m/github-push-action@v0.8.0
   with:
      repository: ${{ env.dest-repo }}
      branch: ${{ env.dest-branch }}
      directory: ${{ env.docs-path }}
      force: true
```